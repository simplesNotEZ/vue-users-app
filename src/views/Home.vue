<template>
  <div class="home">
    <img alt="cloud logo" class="cloud-img" src="../assets/cloud.jpeg" />
    <UsersPanel :users="users" />
  </div>
</template>

<script>
// @ is an alias to /src
import UsersPanel from "@/components/UsersPanel.vue";

import { mapState } from 'vuex';

export default {
  name: "Home",
  components: {
    UsersPanel
  },
  created() {
    if (!this.users) {
      this.$store.dispatch('getUsers');
    }
    // TODO: Potentially delete 
    // api.getById(5)
    //   .then(res => {
    //     console.log("%c res.data from axios.getById: ", "color: fuchsia;", res.data);
    //     this.users =  [res.data];
    //   });
  },
  computed: {
    ...mapState(['users'])
  }
};

</script>

<style scoped>
.cloud-img {
  margin-bottom: 2rem;
  height: 200px;
}
</style>
