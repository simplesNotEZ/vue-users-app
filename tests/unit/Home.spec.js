import { shallowMount } from "@vue/test-utils";
import Home from "@/views/Home.vue";

describe("Home.vue", () => {
  it("renders the stubbed child component", () => {
    const wrapper = shallowMount(Home);
    // console.log('Home wrapper: ', wrapper);
    const imgElement = wrapper.find('img');
    expect(imgElement.attributes('alt')).toBe('cloud logo');
    expect(wrapper.find('userspanel-stub').exists()).toBe(true);
  });

  it("renders correctly", () => {
    const wrapper = shallowMount(Home);
    expect(wrapper).toMatchSnapshot();
  });

  //TODO: add test for mock api GET in created life-cycle hook
  //TODO: add test for emitted event--userDeleted
});

// describe("HelloWorld.vue", () => {
//   it("renders props.msg when passed", () => {
//     const msg = "new message";
//     const wrapper = shallowMount(HelloWorld, {
//       propsData: { msg },
//     });
//     expect(wrapper.text()).toMatch(msg);
//   });
// });
