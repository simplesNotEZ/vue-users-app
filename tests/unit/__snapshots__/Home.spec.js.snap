// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<div class="home"><img alt="cloud logo" src="../assets/cloud.jpeg" height="200" class="cloudImg">
  <userspanel-stub users=""></userspanel-stub>
</div>
`;
